runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java

```diff
@@ -131,39 +142,105 @@ private static String htmlLabel(String key, long value)
 		return String.format(HTML_LABEL_TEMPLATE, ColorUtil.toHexColor(ColorScheme.LIGHT_GRAY_COLOR), key, valueStr);
 	}
 
-	void addLog(final String eventName, final int actorLevel, LootTrackerItemEntry[] items)
+	void addEntry(final String eventName, final int actorLevel, LootTrackerItemEntry[] items, boolean groupLoot)
+	{
+		final String subTitle = actorLevel > -1 ? "(lvl-" + actorLevel + ")" : "";
+
+		LootTrackerEntry entry = new LootTrackerEntry(eventName, subTitle, System.currentTimeMillis(), items);
+		entries.add(entry);
+
+		if (!groupLoot || !stackEntry(entry))
+		{
+			addLog(entry);
+		}
+	}
+
+	private boolean stackEntry(LootTrackerEntry entry)
+	{
+		for (LootTrackerLog log : logs)
+		{
+			if (log.getTitle().equals(entry.getTitle()))
+			{
+				overallGp -= log.getTotalPrice();
+
+				log.addEntry(entry);
+				log.repaint();
+
+				// Update overall
+				overallGp += log.getTotalPrice();
+				overallKills++;
+				updateOverall();
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private void addLog(LootTrackerEntry entry)
 	{
-		// Remove error and show overall
 		remove(errorPanel);
 		overallPanel.setVisible(true);
 
-		// Create box
-		final String subTitle = actorLevel > -1 ? "(lvl-" + actorLevel + ")" : "";
-		final LootTrackerBox box = new LootTrackerBox(itemManager, eventName, subTitle, items);
-		logsContainer.add(box, 0);
-		logsContainer.repaint();
+		LootTrackerLog log = buildLog(entry);
 
 		// Update overall
-		overallGp += box.getTotalPrice();
+		overallGp += log.getTotalPrice();
 		overallKills += 1;
 		updateOverall();
 
+		logs.add(log);
+		logsContainer.add(log, 0);
+	}
+
+	void rebuildLogs(boolean groupLoot)
```

Comment:
rename this to refresh, remove method param

Reply:
You sure? It is only used to re-construct the logs when the grouping config changes (from single to grouped or vice versa), not to refresh any view, rebuild sounds more appropriate

Reply:
then just rebuild i guess